app/lib/shared/redis_cache.dart

```diff
@@ -223,3 +225,57 @@ Future _withInmemoryCache(FutureOr Function() fn) async {
   _registerCache(CachePatterns._(Cache(Cache.inMemoryCacheProvider(4096))));
   return await fn();
 }
+
+typedef CacheProviderFn<T> = Future<CacheProvider<T>> Function();
+
+class _UpdatingCacheProvider<T> implements CacheProvider<T> {
+  final CacheProviderFn<T> _fn;
+  CacheProvider<T> _delegate;
+  Timer _timer;
+  bool _isClosed = false;
+
+  _UpdatingCacheProvider._(this._fn, this._delegate) {
+    _timer = Timer.periodic(Duration(hours: 1), (_) async {
+      _log.info('Starting to update redis connection...');
+      try {
+        final newProvider = await _fn();
+        // Warm-up redis connection
+        await newProvider.get('ping');
+
+        if (_isClosed) {
+          // discard if `close` was called during the Timer callback
+          await newProvider.close();
+        } else {
+          // otherwise update provider and close old one
+          final d = _delegate;
+          _delegate = newProvider;
+          await d.close();
+        }
+      } catch (e, st) {
+        _log.warning('Failed to update redis connection.', e, st);
```

Comment:
Make this a `shout` so it goes into our error reporting? Unless we know it's harmless we can later reduce the noise level.